dash_labs/plugins/pages.py

```diff
@@ -398,3 +430,24 @@ def _parse_query_string(search):
 
         parsed_qs[k] = first
     return parsed_qs
+
+
+def _parse_path_variables(pathname, path_template):
+    path_segments = pathname.split("/")
+    template_segments = path_template.split("/")
+
+    if len(path_segments) != len(template_segments):
+        return None
+
+    path_vars = {}
+    for i in range(len(path_segments)):
+        if path_segments[i] == template_segments[i]:
+            continue
+        if template_segments[i].startswith("<"):
+            # removes the "<" ">" around the variable key
+            variable_key = template_segments[i][1:-1]
+            path_segments[i] = None if path_segments[i] == "None" else path_segments[i]
```

Comment:
This `"None"` -> `None` conversion worries me - what if you really did want to pass `"None"` as the parameter? Can we just leave it as `"None"`? I know up [above](https://github.com/plotly/dash-labs/pull/61/files#diff-3242435debccc1e6c9e9164d3826cd066d02302d12cda2cb20956e71d27da55eR247) we use `"None"` in the path when registering the page but I'm not sure this is a good idea either - for example, how is that going to work in the title, description, and image?

https://github.com/plotly/dash-labs/blob/f182bc011291fedce47c13e6a889d09b17a7c020/dash_labs/plugins/pages.py#L324-L332

I suspect the better way to handle this will be to take the [path-to-layout finder](https://github.com/plotly/dash-labs/pull/61/files#diff-3242435debccc1e6c9e9164d3826cd066d02302d12cda2cb20956e71d27da55eR282-R292) and turn it into a path-to-page finder, then use that both here and in `interpolate_index`

Reply:
In fact, I could imagine users wanting title, description, and image to each potentially be a function of the path variables... But for now we should just ensure they work correctly as constant values for each registered page.

Reply:
Actually this works without any changes.  I updated the example app to include `title` and `description`.  Here's what the meta tags look like when no initial `path` is provided:
 
![image](https://user-images.githubusercontent.com/72614349/145596467-0435fa13-2311-4b97-82fa-3ef9825076d1.png)

And here is what it looks like when a different (valid) initial path is provided:
![image](https://user-images.githubusercontent.com/72614349/145596883-cc920c4d-a326-4775-941e-44952c7352ee.png)

Reply:
> In fact, I could imagine users wanting title, description, and image to each potentially be a function of the path variables... But for now we should just ensure they work correctly as constant values for each registered page.

I think one way to do this is to create a `dash.path_registry`.  It could start by being generated from `dash.page_registry`  Key would be the `path`, values would include `title, description, image`.  Then people could add other paths.

Reply:
>I suspect the better way to handle this will be to take the path-to-layout finder and turn it into a path-to-page finder, then use that both here and in interpolate_index.

Good suggestion - I added a `path_to_page` function in  [this commit](https://github.com/plotly/dash-labs/pull/61/commits/2ddad2ca6f89c45e35fc05fd90139d5cd4276c96)